src/main/java/io/vertx/gsoc2018/qotd/QuoteOfTheDayVerticle.java

```diff
@@ -33,13 +42,66 @@ public void start(Future<Void> startFuture) throws Exception {
       .subscribe(res -> {
         if(res) {
           LOGGER.info("Initialized database successfully");
-          startFuture.complete();
+          startServer(startFuture);
         } else {
           LOGGER.error("Error in initializing database");
         }
       }, err -> LOGGER.error("Error in initializing database: " + err.getMessage()));
   }
 
+  public void startServer(Future<Void> startFuture) {
+    Router router = Router.router(vertx);
+    router.route().handler(BodyHandler.create());
+    router.get("/quotes").handler(this::getQuotesHandler);
+    router.post("/quotes").handler(this::addQuoteHandler);
+
+    vertx.createHttpServer()
+      .requestHandler(router::accept).rxListen(port)
+      .subscribe(res -> {
```

Comment:
Instead of subscribing here, I would return the Single and concat it with the database server setup. Then the server would not be started if the db client fails.

Reply:
Will change it 👍